python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -112,21 +113,16 @@ def __new__(
 
 
 @whitelist_for_serdes
-class RunRequest(
-    NamedTuple(
-        "_RunRequest",
-        [
-            ("run_key", PublicAttr[Optional[str]]),
-            ("run_config", PublicAttr[Mapping[str, Any]]),
-            ("tags", PublicAttr[Mapping[str, str]]),
-            ("job_name", PublicAttr[Optional[str]]),
-            ("asset_selection", PublicAttr[Optional[Sequence[AssetKey]]]),
-            ("stale_assets_only", PublicAttr[bool]),
-            ("partition_key", PublicAttr[Optional[str]]),
-            ("asset_check_keys", PublicAttr[Optional[Sequence[AssetCheckKey]]]),
-        ],
-    )
-):
+@record_custom
+class RunRequest(IHaveNew):
+    run_key: Optional[str]
+    run_config: Optional[Mapping[str, Any]]
+    tags: Mapping[str, str]
+    job_name: Optional[str]
+    asset_selection: Optional[Sequence[AssetKey]]
+    stale_assets_only: bool
+    partition_key: Optional[str]
+    asset_check_keys: Optional[Sequence[AssetCheckKey]]
```

Comment:
@alangenfeld since i need to use `@custom_record` because we do some normalization of `tags` and  `run_config` in `__new__` do i need to keep all of the manual check calls for the other params as well?

Reply:
you can let the generated base one handle all the type checks and remove them from this layer

Reply:
"this layer" being the `__new__` method? or the annotations?

Reply:
ah sorry, the `check` calls inside of the `__new__` impl here can be dropped. We need to keep both the record field type annotations to define the record object and the generated new impl, and the type annotations on `__new__` here to define what types are used for constructing this customized record (even though in this case they are the same we still need to explicitly define them).